features/enlinkd/config/src/main/java/org/opennms/netmgt/config/EnhancedLinkdConfigFactory.java

```diff
@@ -100,14 +89,12 @@ public void reload() throws IOException {
         getWriteLock().lock();
         try {
             final File cfgFile = ConfigFileConstants.getFile(ConfigFileConstants.ENLINKD_CONFIG_FILE_NAME);
-           LOG.debug("init: config file path: {}", cfgFile.getPath());
-            InputStream stream = null;
-            try {
-                stream = new FileInputStream(cfgFile);
-                reloadXML(stream);
-            } finally {
-                if (stream != null) {
-                    IOUtils.closeQuietly(stream);
+            LOG.debug("init: config file path: {}", cfgFile.getPath());
+            try (final InputStream stream = new FileInputStream(cfgFile)){
+                try(final Reader reader = new InputStreamReader(stream)) {
+                    m_config = JaxbUtils.unmarshal(EnlinkdConfiguration.class, reader);
+                } finally {
```

Comment:
@rssntn67 look better :-) I think you can also remove the finally block since it will be auto closed